crates/estree/src/util.rs

```diff
@@ -0,0 +1,97 @@
+use serde_json::Value;
+use serde::ser::*;
+
+pub enum Container {
+    NothingMeaningful,
+    FunctionDeclaration,
+    FunctionExpression,
+}
+
+/// A container used to represent some piece of data being
+/// serialized. Future versions may add options to the
+/// serialization format.
+pub struct Serialization<'a, T> where T: 'a {
```

Comment:
There are two problems. First, I don't think that you can simply implement `serde::ser::Serialize` for `easter` in `estree`. Unless the rules have changed, you can only implement an external trait for a type you have defined locally.

The second reason is more of an ulterior motive. There are at least two widely-used versions of ESTree available: Esprima's and Babel's. Ideally, it would be nice to be able to export to either. For this purpose, we need a way to pass options during serialization. Since Serde doesn't support passing options out of the box, the idea would be to pass them as a field of `Serialization`.